### PR TITLE
Explain when the various model input types can be used

### LINKED ISSUE #5182: [FEATURE][processing] Port multi-layer input parameter widget to new API
Original commit: https://github.com/qgis/QGIS/commit/e386573c0c0af1fa1a328fa92371873c11d04586 by nyalldawson

Allows
- expression based inputs
- rearrangement of inputs for multi-layer parameters in models, to set
specific layer orders

### LINKED ISSUE #5206: [FEATURE][processing] Port output parameter wrappers to new c++ API for modeler
Original commit: https://github.com/qgis/QGIS/commit/d20a3bddc01f334b7fa31ec5417945d8c919e413 by nyalldawson

This allows a range of new possibilities, including:
- models with static outputs for child algorithms, e.g. always saving
a child algorithm's output to a geopackage or postgres layer
- models with expression based output values for child algorithms, e.g.
generating an automatic file name based on today's date and saving
outputs to that file

### PR DESCRIPTION
**Pull request by DelazJ**
and what they provide
Fixes #5206, fixes #5182

### CODE REVIEW

File: docs/user_manual/processing/modeler.rst [hunk 1]
```diff
@@ -396,21 +394,32 @@ As you can see, there are however some differences.
...
+  allows to use the output of another algorithm as an input of the current algorithm.
+  As of model inputs, this option will list all the suitable inputs for the parameter.
+* The **output parameter** also has the above options in its drop-down menu:
+
+  * add static outputs for child algorithms,
+    e.g. always saving a child algorithm's output to a predefined geopackage or postgres layer
+  * use an expression based output values for child algorithms,
+    e.g. generating an automatic file name based on today's date and saving outputs to that file
+  * use a model input
+  * use another algorithm output
```
Comment (DelazJ) [inline docs/user_manual/processing/modeler.rst:419]: Any idea/example of why someone would need to use these options for an output parameter?

Comment (pathmapper) [reply docs/user_manual/processing/modeler.rst:419]: > use a model input

e.g. the `File/Folder` model input to specify an output file or folder

> use another algorithm output

e.g. the output of the `Create directory` algorithm (Modeler tools)

Comment (pathmapper) [inline docs/user_manual/processing/modeler.rst:419]:
```suggestion
  * use a model input,
    e.g. the *File/Folder* model input to specify an output file or folder
  * use another algorithm output,
    e.g. the output of the *Create directory* algorithm (from *Modeler tools*)
```

Comment (DelazJ) [reply docs/user_manual/processing/modeler.rst]: Indeed...

**Pull request merged**